src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/DataSetVersionService.cs

```diff
@@ -100,6 +123,115 @@ private static DataSetVersionSummaryViewModel MapDraftSummaryVersion(DataSetVers
             Type = dataSetVersion.VersionType,
         };
     }
+
+    private async Task<Either<ActionResult, DataSetVersion>> GetDataSetVersion(
+        Guid dataSetVersionId,
+        CancellationToken cancellationToken)
+    {
+        return await publicDataDbContext.DataSetVersions
+            .AsNoTracking()
```

Comment:
Ah, I did this because `GetDataSetVersion` could, in theory, be used by other things - and in those scenarios you may only need to read the data so don't need the tracking. And then here (which, granted, is the only use case for now) I've explicitly started tracking inside the update method.

Do you think it would be better to track it from the beginning instead?

Reply:
Good to know, thanks. I think it's fine to leave it like this. 🙂